ProcessLib/Process.h

```diff
@@ -143,15 +144,21 @@ class Process
 
 private:
 	/// Creates mesh subsets, i.e. components, for given mesh.
-	void initializeMeshSubsets()
+	void initializeMeshSubsets(ProcessVariable const& variable)
 	{
 		// Create single component dof in every of the mesh's nodes.
 		_mesh_subset_all_nodes =
 		    new MeshLib::MeshSubset(_mesh, &_mesh.getNodes());
 
 		// Collect the mesh subsets in a vector.
-		_all_mesh_subsets.push_back(
-		    new MeshLib::MeshSubsets(_mesh_subset_all_nodes));
+		std::generate_n(
+		    std::back_inserter(_all_mesh_subsets),
+		    variable.getNumberOfComponents(),
+		    [&]()
+		    {
+			    return new MeshLib::MeshSubsets(_mesh_subset_all_nodes);
```

Comment:
But currently obviously `_all_mesh_subsets` owns them. So in the simplest case it is the question if you want to change the type of `_all_mesh_subsets` to `std::vector<std::unique_ptr<...>>`. Of course, then you also have to change some method signatures accordingly. Later on the meshsubsets may be owned by the DOF table. I don't care for now.
The problem I see is that most algorithms don't care if they work on a `std::vector<T*>` or on a `std::vector<std::unique_ptr<T>>` so maybe we should think about how to handle that without templating everything, but maybe I'm thinking too far.